src/core/public/core_system.ts

```diff
@@ -126,6 +128,20 @@ export class CoreSystem {
     this.coreApp = new CoreApp(this.coreContext);
   }
 
+  private getLoadMarksInfo() {
+    if (!performance) return [];
+    const reportData = {};
+    performance.mark(KBN_LOAD_MARKS, {
+      detail: 'start_done',
+    });
+    const marks = performance.getEntriesByName(KBN_LOAD_MARKS);
+    for (const mark of marks) {
+      reportData[mark.detail] = mark.startTime;
+    }
+
+    return reportData;
+  }
```

Comment:
This method is failing to compile in TS because `reportData` is declared as `any`.

```suggestion
  private getLoadMarksInfo() {
    if (!performance) return [];
    const reportData: : { [mark: string]: number } = {};
    performance.mark(KBN_LOAD_MARKS, {
      detail: 'start_done',
    });
    // Casting to `PerformanceMark` because `getEntriesByName` returns the base `PerformanceEntry` that does not include the `detail` property.
    // Also adding 'mark' as the type to be more explicit about it being marks.
    const marks = performance.getEntriesByName(KBN_LOAD_MARKS, 'mark') as PerformanceMark[];
    for (const mark of marks) {
      reportData[mark.detail] = mark.startTime;
    }

    return reportData;
  }
```